### PR TITLE
feat: Merge dtype and field format to improve feature saving

### PR DESCRIPTION
**Pull request by cpvannier**
## Description

- **Breaking change:** Update ItemFeature dtype to use the same keywords as Fields:
  - `text` becomes `str`
  - `boolean` becomes `bool`
  - `number` becomes either `int` or `float`
- Update saving methods thanks to this change to better process incoming features
  - Add fields_to_pyarrow and fields_to_python methods for easy switch with string dtypes
  - Update DatasetItem and ItemObject methods
  - Add update_table method in Dataset to automatically add new features

### CODE REVIEW

**Comment by cpvannier** [comment]
Looks functional, but there is probably some debugging left to do.
Will test this more in depth tomorrow.

File: ui/components/core/src/lib/types/objectTypes.ts [hunk 1]
```diff
@@ -4,13 +4,13 @@ import type { BBox, Mask } from "./datasetTypes";
 export type ObjectParameters = {
```
Comment (timothee-LJN) [inline ui/components/core/src/lib/types/objectTypes.ts]: This type in unused, removed in commit fbf3766

File: ui/components/imageWorkspace/src/components/Features/TextInputFeature.svelte [hunk 2]
```diff
@@ -18,30 +18,53 @@
 
   import { Input } from "@pixano/core/src";
 
-  import type { TextFeature, NumberFeature } from "../../lib/types/imageWorkspaceTypes";
+  import type { TextFeature, IntFeature, FloatFeature } from "../../lib/types/imageWorkspaceTypes";
```
Comment (timothee-LJN) [inline ui/components/imageWorkspace/src/components/Features/TextInputFeature.svelte:21]: To keep it simpler, you could keep a type : 
```
type NumberFeature = IntFeature | FloatFeature
```

File: ui/components/imageWorkspace/src/components/Features/TextInputFeature.svelte [hunk 3]
```diff
@@ -18,30 +18,53 @@
...
 
-  export let textFeature: Pick<NumberFeature | TextFeature, "name" | "value">;
+  export let textFeature: Pick<IntFeature | FloatFeature | TextFeature, "name" | "value">;
   export let isEditing: boolean;
   export let saveInputChange: (value: string | number, propertyName: string) => void;
-  export let inputType: "text" | "number" = "text";
+  export let inputType: "str" | "int" | "float" = "str";
```
Comment (timothee-LJN) [inline ui/components/imageWorkspace/src/components/Features/TextInputFeature.svelte:26]: I would suggest here to use created type : 
```
export let inputType: NumberFeature["type"] | TextFeature["type"] = "str";
```

File: ui/components/imageWorkspace/src/components/Features/TextInputFeature.svelte [hunk 4]
```diff
@@ -18,30 +18,53 @@
...
 
 <div class="flex justify-start items-center gap-4">
   {#if isEditing}
-    <Input
-      value={textFeature.value}
-      on:change={(e) => onTextInputChange(e.currentTarget.value, textFeature.name)}
-      on:input={() => (isSaved = false)}
-      type={inputType}
-    />
+    {#if inputType === "str"}
```
Comment (timothee-LJN) [inline ui/components/imageWorkspace/src/components/Features/TextInputFeature.svelte:44]: Here is as suggested simplification : 
```
<Input
      value={textFeature.value}
      on:change={(e) => onTextInputChange(e.currentTarget.value, textFeature.name)}
      on:input={() => (isSaved = false)}
      type={inputType === "str" ? "text" : "number"}
      step={inputType === "int" ? "1" : "any"} 
    />
```

`step=any` would not change anything if input type is text !

File: ui/components/imageWorkspace/src/components/SaveShape/SaveShapeForm.svelte [hunk 5]
```diff
@@ -155,6 +155,22 @@
           </span>
           <Input
             type="number"
+            step="1"
```
Comment (timothee-LJN) [inline ui/components/imageWorkspace/src/components/SaveShape/SaveShapeForm.svelte:158]: I guess you could use the same code and change only the step depending on type `int` or `float`

File: ui/components/imageWorkspace/src/lib/types/imageWorkspaceTypes.ts [hunk 6]
```diff
@@ -25,25 +25,30 @@ export type CreateObjectSchemaDefinition = Record<string, z.ZodTypeAny>;
...
+  type: "float";
   value: number;
 };
 
 export type ListFeature = ListInput & {
   value: string;
 };
 
-export type Feature = CheckboxFeature | TextFeature | NumberFeature | ListFeature;
+export type Feature = CheckboxFeature | TextFeature | IntFeature | FloatFeature | ListFeature;
```
Comment (timothee-LJN) [inline ui/components/imageWorkspace/src/lib/types/imageWorkspaceTypes.ts:51]: as mentioned above, keeping a `NumberFeature` type could be useful!

**Pull request merged**